providers/dns/nearlyfreespeech/nearlyfreespeech.toml

```diff
@@ -0,0 +1,25 @@
+Name = "nearlyfreespeech.net"
+Description = ''''''
+URL = "https://nearlyfreespeech.net/"
+Code = "nearlyfreespeech"
+Since = "v0.1.0"
+
+Example = '''
+NFS_API_KEY=xxxxxx \
+NFS_LOGIN=xxxx \
+lego --email myemail@example.com --dns nearlyfreespeech --domains my.example.org run
+'''
+
+[Configuration]
+  [Configuration.Credentials]
+    NFS_API_KEY = "API Key for API requests"
+    NFS_LOGIN = "Username for API requests"
+  [Configuration.Additional]
+    NFS_POLLING_INTERVAL = "Time between DNS propagation check"
+    NFS_PROPAGATION_TIMEOUT = "Maximum waiting time for DNS propagation"
+    NFS_TTL = "The TTL of the TXT record used for the DNS challenge"
+    NFS_HTTP_TIMEOUT = "API request timeout"
+    NFS_SEQUENCE_INTERVAL = "Time between sequential requests"
+
+[Links]
+  API = "https://api.nearlyfreespeech.net"
```

Comment:
this must be a link to the public documentation of the API.

Reply:
There is no publicly hosted version of their docs right now that I've been able to find. Is that going to be a nonstarter?

Reply:
Yes, it's a problem: we cannot maintain something without public API documentation.

How did you create the client implementation (what is your API reference/documentation)?

Reply:
I have an account with NFS and the API documentation is available at that URL when you login.

Reply:
ok I will try to take a look.

Reply:
This is the only unresolved review here and I understand the issue that maintaining this is hard when there is no public documentation. Is that an absolute blocker when there is pretty decent API documentation available after login or is it being public a hard requirement that will keep this from getting merged?

Reply:
I was able to create a "free" account and access the documentation.
I don't understand why this documentation is not publicly available.

The API is simple, not well documented but simple, and it's possible to access the documentation without paying something, so I think we can add this provider.